### PR TITLE
Allowing override of http transport in credentials factory.

### PR DESCRIPTION
**Pull request by sduskis**
Some users want to set a proxy.  This will allow users to do so.

### CODE REVIEW

**Comment by igorbernstein2** [comment]
This feels a bit weird, shouldn't this be configured via BigtableOptions?

**Comment by sduskis** [comment]
We have some caching of credentials.  Allowing this to be set via CredentialsOptions would require us to remove that cache.  This is a stop gap solution.

The best answer here might be JWT credentials, which makes it so that we don't have to do remote calls at all.

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java [hunk 1]
```diff
@@ -79,16 +79,35 @@
...
   // credential servers. Initialized lazily to move the possibility of throwing
   // GeneralSecurityException to the time a caller actually tries to get a credential.
-  private static HttpTransportFactory httpTransportFactory = new HttpTransportFactory() {
-    @Override
-    public HttpTransport create() {
-      try {
-        return GoogleNetHttpTransport.newTrustedTransport();
-      } catch (Exception e) {
-        throw new RuntimeException("Could not create a GoogleNetHttpTransport: " + e.getMessage(), e);
+  private static HttpTransportFactory httpTransportFactory;
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java]: This needs to be volatile as well:

http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html

Comment (sduskis) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java]: I removed the synchronized check as per our offline conversation.

**Comment by igorbernstein2** [comment]
LGTM

**Pull request merged**